atomix/cluster/src/main/java/io/atomix/raft/storage/log/RaftLogUncommittedReader.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2017-present Open Networking Foundation
+ * Copyright © 2020 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.atomix.raft.storage.log;
+
+import io.atomix.raft.storage.log.entry.RaftLogEntry;
+import io.atomix.raft.storage.serializer.RaftEntrySBESerializer;
+import io.atomix.raft.storage.serializer.RaftEntrySerializer;
+import io.camunda.zeebe.journal.JournalReader;
+import io.camunda.zeebe.journal.JournalRecord;
+import java.util.NoSuchElementException;
+
+/**
+ * Raft log reader that reads both committed and uncommitted entries. This reader is supposed to be
```

Comment:
:thought_balloon: I wonder whether there is a way to make it impossible to access this class from outside Raft. E.g. by introducing an interface for `RaftLog` which doesn't contain the `openReader()` method. Then internally, we could use the implementation class, but external parties only see the method to get a committed reader.

Reply:
The API exposed via `RaftPartitionServer` only can open a committedReader. This is what external services use. Is that enough?